sktime/transformations/series/boxcox.py

```diff
@@ -158,18 +165,63 @@ def _boxcox(x, lmbda=None, bounds=None, alpha=None):
 class BoxCoxTransformer(_SeriesToSeriesTransformer):
     _tags = {"transform-returns-same-time-index": True, "univariate-only": True}
 
-    def __init__(self, bounds=None, method="mle"):
+    def __init__(self, bounds=None, method="mle", sp=None):
         self.bounds = bounds
         self.method = method
         self.lambda_ = None
+        self.sp = sp
         super(BoxCoxTransformer, self).__init__()
 
     def fit(self, Z, X=None):
+        self._check_method()
         z = check_series(Z, enforce_univariate=True)
-        self.lambda_ = _boxcox_normmax(z, bounds=self.bounds, method=self.method)
+        if self.method != "guerrero":
+            self.lambda_ = _boxcox_normmax(z, bounds=self.bounds, method=self.method)
+        else:
+
+            def _guerrero(x, sp):
```

Comment:
I'd prefer adding this to the above `_boxcox_normmax` function as a possible input for the `method` argument. We could even ask SciPy if they'd be interested in including it.

Reply:
That way, we can avoid the if-else clause and also get rid of the `self._check_method` helper function, as this would move inside the `_boxcox_normmax` function

Reply:
We also have a helper function `check_sp` in `sktime/utils/validation/forecasting` I believe

Reply:
Thanks for the review, @mloning!
I moved `_guerrero` definition from `_boxcox_normmax` to the `fit` method of `BoxCoxTransformer` because of how I understood [this comment](https://github.com/alan-turing-institute/sktime/pull/791#pullrequestreview-632255503), did I get it wrong?
So it's better if I move the method back inside  `_boxcox_normmax` and pass `sp` there as a parameter? I personally agree that `_guerrero` fits better in there.

Another question is about a test against results from R's implementation: I'm not quite sure how to get these results. Is there a way to run R code here in sktime or sklearn? Or do you mean hard-coded values taken from running R code locally and putting them into the test?

Reply:
Oh yes, I forgot about that comment, probably makes sense to keep it separate, but I would still define the `_guerrero(...)` function outside of the `fit` function as a standalone function and pass arguments as necessary inside fit. That would make the if-else clause more readable.

Reply:
Yes hard-coded values would be fine I think, perhaps pasting the R code to generate the values as a comment.

Reply:
Hi @mloning! 
I've just moved it outside the `fit` as a standalone function and added testing against R implementation with hardcoded values.